spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc

```diff
@@ -0,0 +1,223 @@
+* link:https://github.com/spring-cloud/spring-cloud-stream-app-starters/tree/master/metrics/spring-cloud-starter-stream-sink-aggregate-counter[Aggregate Counter] - Stores total counts but also retains the total count values for each minute, hour day and month.
+
+It is important to note that the timestamp that is used in the aggregate counter can come from a field in the message itself so that out of order messages are properly accounted.
+
+ifdef::omit-tasks-docs[]
+[[arch-task]]
+== Task Applications
+
+The Spring Cloud Task programming model provides 
+
+* Persistence of the Task’s lifecycle events and exit code status.
+* Lifecycle hooks to execute code before or after a task execution.
+* Emit task events to a stream (as a source) during the task lifecycle.
+* Integration with Spring Batch Jobs.
+endif::omit-tasks-docs[]
+
+[[arch-data-flow-server]]
+== Data Flow Server
+
+[[arch-data-flow-server-endpoints]]
+=== Endpoints
+
+The Data Flow Server uses an embedded servlet container and exposes REST endpoints for creating, deploying, undeploying, and destroying streams and tasks, querying runtime state, analytics, and the like. The Data Flow Server is implemented using Spring’s MVC framework and the link:https://github.com/SpringSource/spring-hateoas[Spring HATEOAS] library to create REST representations that follow the HATEOAS principle.
+
+.The Spring Cloud Data Flow Server 
+image::{dataflow-asciidoc}/images/data-flow-server-arch.png[The Spring Cloud Data Flow Server Architecture, scaledwidth="100%"]
+
+[[arch-data-flow-server-customization]]
+=== Customization
+
+Each Data Flow Server executable jar targets a single runtime by delegating to the implementation of the deployer Service Provider Interface found on the classpath.  
+
+We provide a Data Flow Server executable jar that targets a single runtime.  The Data Flow server delegates to the implementation of the deployer Service Provider Interface found on the classpath.  In the current version, there are no endpoints specific to a target runtime, but may be available in future releases as a convenience to access runtime specific features 
+
+While we provide a server executable for each of the target runtimes you can also create your own customized server application using Spring Initialzr.   This let’s you add or remove functionality relative to the executable jar we provide.  For example, adding additional security implementations, custom endpoints, or removing Task or Analytics REST endpoints.  You can also enable or disable some features through the use of feature toggles.
+
+[[arch-data-flow-server-security]]
+=== Security
+
+The Data Flow Server executable jars support basic http and OAuth 2.0 authentication to access it endpoints.  Refer to the security section for more information.  
+
+Authorization via groups is planned for a future release.
+
+[[arch-runtime]]
+== Security
+
```

Comment:
Missing test for Security i.e.
The Data Flow Server executable jars support basic http and OAuth 2.0 authentication to access it endpoints.  Refer to the security section for more information.  

Authorization via groups is planned for a future release.